test/test-moveTetromino.js

```diff
@@ -0,0 +1,829 @@
+            -----
+            --XX-
+            ---X-
+            ---X-
+        `)
+        let field = tetris.buildField(`
+            -----
+            -----
+            -----
+            -----
+        `);
+        let shift = { x: 0, y: -1 };
+        let lTetromino = new tetris.Tetromino(typeOfTetromino, phase, coordinatesTetromino);
+        let resultMoveTetromino = lTetromino.moveTetromino(field, shift);
+        let expected = tetris.getCoordinates(`
+            -----
+            --XX-
+            ---X-
+            ---X-
+        `);
+    })
+    it("Move down J tetromino", function () {
+        let typeOfTetromino = "J";
+        let phase = "up";
+        let coordinatesTetromino = tetris.getCoordinates(`
+            -XX--
+            -X---
+            -X---
+            -----
+        `)
+        let field = tetris.buildField(`
+            -----
+            -----
+            -----
+            -----
+        `);
+        let shift = { x: 0, y: -1 };
+        let jTetromino = new tetris.Tetromino(typeOfTetromino, phase, coordinatesTetromino);
+        let resultMoveTetromino = jTetromino.moveTetromino(field, shift);
+        let expected = tetris.getCoordinates(`
+            -----
+            -XX--
+            -X---
+            -X---
+        `)
+    })
+    it("J tetromino can not move, there is an element on the way", function () {
+        let typeOfTetromino = "J";
+        let phase = "up";
+        let coordinatesTetromino = tetris.getCoordinates(`
+            -XX--
+            -X---
+            -X---
+            -----
+        `)
+        let field = tetris.buildField(`
+            -----
+            --X--
+            -----
+            -----
+        `);
+        let shift = { x: 0, y: -1 };
+        let jTetromino = new tetris.Tetromino(typeOfTetromino, phase, coordinatesTetromino);
+        let resultMoveTetromino = jTetromino.moveTetromino(field, shift);
+        let expected = tetris.getCoordinates(`
+            -XX--
+            -X---
+            -X---
+            -----
+        `)
+    })
+    it("J tetromino can not move, item too left", function () {
+        let typeOfTetromino = "J";
+        let phase = "down";
+        let coordinatesTetromino = tetris.getCoordinates(`
+            -X---
+            -X---
+            XX---
+            -----
+        `)
+        let field = tetris.buildField(`
+            -----
+            -----
+            -----
+            -----
+        `);
+        let shift = { x: -1, y: 0 };
+        let jTetromino = new tetris.Tetromino(typeOfTetromino, phase, coordinatesTetromino);
+        let resultMoveTetromino = jTetromino.moveTetromino(field, shift);
+        let expected = tetris.getCoordinates(`
+            -X---
+            -X---
+            XX---
+            -----
+        `)
+    })
+    it("J tetromino can not move, item too right", function () {
+        let typeOfTetromino = "J";
+        let phase = "down";
```

Comment:
moving down should be possible in this case

Reply:
but I move tetromino left

Reply:
but I move tetromino right